### ISSUE TITLE
Error not reported when using bad API key

### ISSUE THREAD

**Issue by Swimburger** [AssemblyAI/assemblyai-java-sdk#63]
When you use the realtime transcriber with an empty API key, the error is not passed to `onError`.

I am getting the following output in the console.

```json
{
  "message_type" : "SessionBegins",
  "error" : "Not authorized"
}
```

and 

```json
{
  "session_id" : "ab0ae487-ed4d-4d24-b6c3-65a701df20f6",
  "expires_at" : "2024-01-18T21:31:50.915638Z",
  "message_type" : "SessionBegins"
}
{
  "message_type" : "SessionBegins",
  "error" : "Received invalid request. Please check the documentation for the correct request schemas."
}
```

This is only in the Java SDK, so it's not how the service is sending the message to the client, but how the Java SDK is deserializing the message.

**Comment by Swimburger** [comment]
I also get the same type of message even after the session has begun:

```json
{
  "session_id" : "ab0ae487-ed4d-4d24-b6c3-65a701df20f6",
  "expires_at" : "2024-01-18T21:31:50.915638Z",
  "message_type" : "SessionBegins"
}
{
  "message_type" : "SessionBegins",
  "error" : "Received invalid request. Please check the documentation for the correct request schemas."
}
```

**Comment by Swimburger** [comment]
Fixed as part of #101.

@dsinghvi FYI, the discriminated union didn't work properly, so I had to do a more manual if/elseif/else.

**Comment by Swimburger** [comment]
This change has been released in [1.1.0](https://central.sonatype.com/artifact/com.assemblyai/assemblyai-java).
